### PR TITLE
Add support for the nrf52840_dongle board.

### PR DESCRIPTION
**Pull request by gendx**
### Pull Request Overview

This pull request adds support for the [nRF52840-Dongle](https://www.nordicsemi.com/Software-and-tools/Development-Kits/nRF52840-Dongle) board. It is similar to the nRF52840-DK, but with a different layout of GPIOs, and available LEDs/buttons.

### Testing Strategy

This pull request was tested by flashing Tock on such a dongle and manually testing simple apps.

### TODO or Help Wanted

N/A

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

I added the relevant README. Is there anything else to update?

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

File: boards/nordic/nrf52840_dongle/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+Platform-Specific Instructions: nRF52840-DK
+===================================
+
+The [nRF52840 Dongle](https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF52840-DK)
```
Comment (bradjc) [inline boards/nordic/nrf52840_dongle/README.md]: Update.

**Comment by alevy** [comment]
@bradjc

> I'm curious about the code loading situation though, what does it take to flash this with JLink?

it looks like this is documented in the README in the PR---there are pads for a TAG connect cable, similar to imix and hail.

> And what would it take to be able to flash it with the USB header?

It also seems like it might be possible to do this using the [nRF52 USB DFU endpoint](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fug_nrf52840_dongle%2FUG%2Fnrf52840_Dongle%2Fgetting_started.html), which might just require compiling the kernel to not overwrite some specific segment of flash or something.

> I'm worried that this board looks very compelling (only $10 on Digikey!) but then if actually loading code on it is very difficult that would be a big hurdle for someone trying to use Tock.

I'm not personally worried about this. I think it's important to have easy to use boards that we direct new people towards, and maybe specifically write our documentation for (e.g. the nRF52 DKs in this case) _as well_ as boards that target specific use cases (e.g. the nrf52 dongle for U2F)

**Comment by bradjc** [comment]
OK, but I made it more explicit. Having the USB connector is just too misleading.

**Comment by gendx** [comment]
The new wording looks clear to me.

**Comment by gendx** [comment]
Fixed the `setup_board` call to sync with other pull-requests that merged in the meantime.

**Pull request merged**